### PR TITLE
EZP-31800: Cleared persistence cache on Content Type copy

### PR DESCRIPTION
**Pull request by Nattfarinn**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31800](https://jira.ez.no/browse/EZP-31800)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.5`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 1]
```diff
@@ -332,8 +332,17 @@ public function createDraft($modifierId, $typeId)
     public function copy($userId, $typeId, $status)
     {
         $this->logger->logCall(__METHOD__, ['user' => $userId, 'type' => $typeId, 'status' => $status]);
+        $copy = $this->persistenceHandler->contentTypeHandler()->copy($userId, $typeId, $status);
 
-        return $this->persistenceHandler->contentTypeHandler()->copy($userId, $typeId, $status);
+        // Clear loadContentTypes() cache as we remove an item from it's collection here.
+        $this->cache->deleteItems(array_map(
+            function ($groupId) {
```
Comment (adamwojs) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php]:
```suggestion
            static function ($groupId): string {
```

+ missing type hint for $groupId

**Review by adamwojs — APPROVED** [review]

**Review by alongosz — APPROVED** [review]

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 2]
```diff
@@ -332,8 +332,17 @@ public function createDraft($modifierId, $typeId)
     public function copy($userId, $typeId, $status)
     {
         $this->logger->logCall(__METHOD__, ['user' => $userId, 'type' => $typeId, 'status' => $status]);
+        $copy = $this->persistenceHandler->contentTypeHandler()->copy($userId, $typeId, $status);
 
-        return $this->persistenceHandler->contentTypeHandler()->copy($userId, $typeId, $status);
+        // Clear loadContentTypes() cache as we remove an item from it's collection here.
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php]: The problem is rather that we add an item right? A remove would have trigger cache clear on the type tag right?

**Review by andrerom — APPROVED** [review]
+1 besides nitpick :)

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 2 above)
Comment (Nattfarinn) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php]: @andrerom Of course you're right. I was thinking if delete an item will have the same issue at the time of writing this. (edit: made message a direct copy of create one)

**Pull request merged**